src/somesy/core/core.py

```diff
@@ -10,32 +10,38 @@ def get_project_metadata(path: Path) -> ProjectMetadata:
     """Return project metadata to from different sources as a simple dict. Currently, only pyproject.toml inputs are supported.
 
     Args:
-        path (Path): config source path
+        path (Path): input source path
 
     Returns:
         ProjectMetadata: project metadata input to sync output files
     """
     # load the pyproject.toml file
     if path.suffix == ".toml":
-        try:
-            with open(path, "r") as f:
-                config = load(f)
-        except FileNotFoundError:
-            raise FileNotFoundError("pyproject.toml not found")
+        with open(path, "r") as f:
+            input = load(f)
 
         # load project metadata
-        if (
-            "tool" in config
-            and "somesy" in config["tool"]
-            and "project" in config["tool"]["somesy"]
+        if "somesy" in path.name and "project" in input:
+            metadata = input["project"]
+            try:
+                metadata = metadata.unwrap()
+                return ProjectMetadata(**metadata)
+            except Exception as e:
+                raise ValueError(f"Somesy input validation failed: {e}")
+
+        elif (
+            "pyproject" in path.name
+            and "tool" in input
+            and "somesy" in input["tool"]
+            and "project" in input["tool"]["somesy"]
         ):
-            metadata = config["tool"]["somesy"]["project"]
+            metadata = input["tool"]["somesy"]["project"]
             try:
                 metadata = metadata.unwrap()
                 return ProjectMetadata(**metadata)
             except Exception as e:
                 raise ValueError(f"Somesy input validation failed: {e}")
         else:
-            raise ValueError("Somesy config not found in pyproject.toml file.")
+            raise ValueError("Somesy input not found.")
     else:
-        raise ValueError("Only toml files are supported for reading project metadata.")
+        raise ValueError("Only toml files are supported for reading somesy input.")
```

Comment:
Phew, this looks very convoluted, and like something you'd need to touch whenever you add a new input source. And its not easy to understand, because all the input types are entangled. I get what you're doing, but I'm not really happy about this. Well, maybe something to keep in mind and refactor in a subsequent iteration.

Reply:
@apirogov what is your recommendation?

Reply:
@apirogov created [this issue](https://github.com/Materials-Data-Science-and-Informatics/somesy/issues/5). I will do it at a later iteration.